### PR TITLE
Add support for running Frontera

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

I haven't worked out how to properly launch the tests, but I would expect `ibrun ./bin/NonFailureTestsRunTests.sh` to work. Someone else can figure that out.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: support/SubmitScripts/Frontera.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+echo ""
+echo ""
+
+export IBRUN_TASKS_PER_NODE=1
+
+if [ -f ${SPECTRE_EXECUTABLE} ]; then
+    if [ -f ${SPECTRE_INPUT_FILE} ]; then
+        ibrun -n ${SLURM_JOB_NUM_NODES} \
+            ${SPECTRE_EXECUTABLE} ++ppn 55 \
+            --input-file ${SPECTRE_INPUT_FILE} > SpECTRE.out 2>&1
```
Comment (kidder) [inline support/SubmitScripts/Frontera.sh]: what goes into SpECTRE.out that does not go into the `-o` and `-e` `SBATCH` options?

**Review by kidder — CHANGES_REQUESTED** [review]

File: support/SubmitScripts/Frontera.sh (hunk 1 above)
Comment (nilsdeppe) [reply support/SubmitScripts/Frontera.sh]: As is nothing from the spectre output goes into `-o` and `-e`. I think this is what SpEC did, though I'm not sure I see any value in it. I do like having all the error and standard output in one file. What do you think of dumping all executable output into `-o`?

Comment (kidder) [reply support/SubmitScripts/Frontera.sh]: that would be fine with me

Comment (nilsdeppe) [reply support/SubmitScripts/Frontera.sh]: Okay, updated and squashed

File: support/SubmitScripts/Frontera.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#SBATCH -n 2               # Total # of tasks, must be number of nodes
+#SBATCH -t 00:30:00        # Run time (hh:mm:ss)
+# #SBATCH --mail-type=all  # Send email at begin and end of job
+# #SBATCH --mail-user=username@tacc.utexas.edu
+# #SBATCH -A account       # Project/Allocation name (req'd if you have more than 1)
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# UPDATE THESE!!!
```
Comment (kidder) [inline support/SubmitScripts/Frontera.sh:31]: hmm, not sure what failed (most likely me...), but I don't see the following request that I attempted to make earlier:
Can you insert here similar documentation as found in `SubmitScripts/Wheeler.sh`

**Review by kidder — CHANGES_REQUESTED** [review]
I am sorry, it appears I had a GitHub failure; this was the change I wanted to request; the comment you just addressed was meant more as a question to discuss...

**Review by kidder — DISMISSED** [review]

**Comment by kidder** [comment]
@geoffrey4444 please look at

**Comment by geoffrey4444** [comment]
I tried building EvolveKerrSchild, following the instructions, and I get this compiler error:

~~~
In file included from /home1/01222/geoffrey/Codes/spectre/src/Time/StepControllers/StepController.hpp:12,
                 from /home1/01222/geoffrey/Codes/spectre/src/Time/StepControllers/SimpleTimes.hpp:16,
                 from /home1/01222/geoffrey/Codes/spectre/src/Time/StepControllers/SimpleTimes.cpp:4:
/home1/01222/geoffrey/Codes/spectre/src/Parallel/PupStlCpp11.hpp:98:13: error: redefinition of ‘template<class K, class V, class H> void PUP::operator|(PUP::er&, std::unordered_map<V, T, Cmp>&)’
   98 | inline void operator|(er& p, std::unordered_map<K, V, H>& m) {  // NOLINT
      |             ^~~~~~~~
In file included from /home1/01222/geoffrey/apps/charm/src/ck-core/sdag.h:46,
                 from /home1/01222/geoffrey/apps/charm/mpi-linux-x86_64-smp-mpicxx/tmp/CkMarshall.decl.h:6,
                 from /home1/01222/geoffrey/apps/charm/src/ck-core/charm++.h:134,
                 from /home1/01222/geoffrey/Codes/spectre/src/Parallel/Abort.hpp:9,
                 from /home1/01222/geoffrey/Codes/spectre/src/ErrorHandling/Assert.hpp:14,
                 from /home1/01222/geoffrey/Codes/spectre/src/Utilities/PointerVector.hpp:13,
                 from /home1/01222/geoffrey/Codes/spectre/build_gcc/SpectrePch.hpp:18:
/home1/01222/geoffrey/apps/charm/src/util/pup_stl.h:259:15: note: ‘template<class V, class T, class Cmp> void PUP::operator|(PUP::er&, std::unordered_map<V, T, Cmp>&)’ previously declared here
  259 |   inline void operator|(er &p,typename std::unordered_map<V,T,Cmp> &m)
      |               ^~~~~~~~
~~~

**Comment by nilsdeppe** [comment]
ah that's because I forgot to apply the patch :(

It sounds like we will merge 6.10.2 tonight/tomorrow so I'll wait until that is done and then update this PR. Is that okay? If you wanted to test more out, you can apply the Charm++ manually for now

**Comment by geoffrey4444** [comment]
After patching charm, it builds fine, except I need to get an interactive job to compile EvolveKerrSchild, because otherwise frontera says I don't have enough memory

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@geoffrey4444 please look at

**Comment by kidder** [comment]
@geoffrey4444 please look at

**Review by geoffrey4444 — APPROVED** [review]

**Pull request merged**